src/sema/types.rs

```diff
@@ -1539,7 +1539,7 @@ impl Type {
     /// Is this a storage bytes string
     pub fn is_storage_bytes(&self) -> bool {
         if let Type::StorageRef(_, ty) = self {
-            if let Type::DynamicBytes = ty.as_ref() {
+            if let Type::DynamicBytes | Type::Bytes(_) = ty.as_ref() {
```

Comment:
I am not sure this is the right solution. I think @seanyoung can analyze this better than me.

Reply:
I agree, I don't think this is the right solution. Having said that, I don't know what is! Let me get back to you on this @xermicus  (I had thought this before but never got round to finding the right solution)

Reply:
This problem is the same as https://github.com/hyperledger/solang/pull/1115 with one additional detail: `fn storage_array_elem()` should return `Type::Bytes(1)`  for `Type::StorageRef(Type::Bytes(_))`.

Reply:
This kind of was my first attempt we scrapped

Reply:
I think that attempt was correct, a fixed bytes is an array type. Sorry only noticed that conversation now.

The other fix that is neeeded is that sema should load the `bytes32` from storage with an `Expression::Load()`.